app/containers/Passcode/PasscodeEnter.tsx

```diff
@@ -23,13 +23,13 @@ const PasscodeEnter = ({ theme, hasBiometry, finishProcess }: IPasscodePasscodeE
 	const ref = useRef(null);
 	let attempts: any = 0;
 	let lockedUntil: any = false;
-	const [passcode, setPasscode] = useState(null);
+	const [passcode, setPasscode] = useState<string | null | undefined>(null);
 	const [status, setStatus] = useState(null);
 	const { getItem: getAttempts, setItem: setAttempts } = useAsyncStorage(ATTEMPTS_KEY);
 	const { setItem: setLockedUntil } = useAsyncStorage(LOCKED_OUT_TIMER_KEY);
 
-	const fetchPasscode = async () => {
-		const p: any = await UserPreferences.getStringAsync(PASSCODE_KEY);
+	const fetchPasscode = () => {
```

Comment:
Great option!! 
Take a look at the implementation:
[useUserPreferences](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3634/commits/38a9d398fe08558cf79b8f4ed85ac6a8ec925c50)
[implementation](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3634/commits/2e6c78cdcaee2ffa23bd61e71604ce601cb7cd43)

Reply:
Coming from Github Discover. You can also use `create` function exposed by the library to build the hook instead of wrapping `useMMKVStorage` hook with another hook. 

```js
const useUserPreferences = create(MMKV, defaultValue)
```

Reply:
@ammarahm-ed  thanks for the tip